src/find_libpython/__init__.py

```diff
@@ -165,6 +182,13 @@ def candidate_names(suffix=_SHLIB_SUFFIX):
     if DLLLIBRARY:
         yield DLLLIBRARY
 
+    if _is_posix:
+        try:
+            yield from _get_proc_library(basename=True)
+        except:
```

Comment:
ruff doesn't like this. I think it'd be better to move this try-except block to directly around the `open` call. try-excepts should be local as possible, and catch as few exceptions as possible, per code quality guidelines.
```suggestion
        except:
```